docs/samples/shards/General/CaptureLog/1.edn

```diff
@@ -0,0 +1,28 @@
+(defloop with-ui
+  (GFX.MainWindow
+   :Title "Console demo" :Width 800 :Height 600
+   :Contents
+   (->
+    (Setup
+     (GFX.DrawQueue) >= .ui-draw-queue
+     (GFX.UIPass .ui-draw-queue) >> .render-steps)
+
+    .ui-draw-queue (GFX.ClearQueue)
+
+    (UI
+     .ui-draw-queue
+     (->
+      (UI.CentralPanel
+       (->
+        (UI.Button "Say Hello" (Msg "Hello!"))
+        (CaptureLog
+         :Size 24
+         :MinLevel "info"
+         :Pattern "%^[%l]%$ [%Y-%m-%d %T.%e] %v")
+        (UI.Console)))))
```

Comment:
how does this work? Shouldn't CaptureLog be BLOCKING (via coroutines)?

Reply:
Why should it be blocking? It uses a ring buffer internally.

Reply:
Added an option to suspend.